cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -229,7 +230,16 @@ runQueryProtocolParameters (AnyConsensusModeParams cModeParams) network mOutFile
           & onLeft (left . ShelleyQueryCmdUnsupportedNtcVersion)
           & onLeft (left . ShelleyQueryCmdEraMismatch)
 
-  writeProtocolParameters mOutFile =<< except (join (first ShelleyQueryCmdAcquireFailure result))
+  except (join (first ShelleyQueryCmdAcquireFailure result))
+
+runQueryProtocolParameters
+  :: AnyConsensusModeParams
+  -> NetworkId
+  -> Maybe OutputFile
+  -> ExceptT ShelleyQueryCmdError IO ()
+runQueryProtocolParameters cModeParams network mOutFile =
+  executeQueryProtocolParameters cModeParams network
+    >>= writeProtocolParameters mOutFile
```

Comment:
This code connects once to get the protocol parameters and again to build the transaction.

What might be involved in making this happen over one connection instead?

Reply:
This will get addressed in the next step, I've added to tasks to original issue #5052. Resolving for now.